src/main/sensors/voltage.c

```diff
@@ -362,3 +363,27 @@ void voltageMeterRead(voltageMeterId_e id, voltageMeter_t *meter)
         voltageMeterReset(meter);
     }
 }
+
+// update voltageStableBits
+// new bit is shifted in every VOLTAGE_STABLE_UPDATE_MS
+// bit 1 is shifted in when voltage did not change more than VOLTAGE_STABLE_MAX_DELTA
+// bit 0 is shifted in othervise
+void voltageStableUpdate(voltageMeter_t* vm)
+{
+    const uint32_t now = millis();
+    if (abs(vm->voltageStablePrevFiltered - vm->displayFiltered) > VOLTAGE_STABLE_MAX_DELTA) {
+        // reset stable voltage reference
+        vm->voltageStablePrevFiltered = vm->displayFiltered;
+        vm->voltageStableBits &= ~BIT(0);  // woltage threshold exceeded in this period
+    }
+    if (cmp32(now, vm->voltageStableLastUpdate) >= VOLTAGE_STABLE_TICK_MS) {
+        vm->voltageStableBits = (vm->voltageStableBits << 1) | BIT(0);  // start with 'stable' state
```

Comment:
is `| BIT(0)` really necessary as << 1 shifts a 0 already?

Reply:
It it bit zero (LSB), not bit with value of 0. New 1 bit is shifted in on each update and possibly cleared above.

Reply:
Makes sense!